include/GoogleSync/GoogleSyncBase.php

```
@@ -0,0 +1,1049 @@
+        $query = "SELECT id FROM meetings WHERE gsync_id = {$eventIdQuoted}";
+        $result = $this->db->query($query);
+        
+        if (!$result) {
+            throw new GoogleSyncException('Meeting not found with specified gsync_id: ' . $eventIdQuoted, GoogleSyncException::MEETING_NOT_FOUND);
+        }
+
+        // This checks to make sure we only get one result. If we get more than one, something is inconsistant in the DB
+        if ($result->num_rows > 1) {
+            // TODO: It shouldn't be in the log as it is an exception, it will be logged if unhandled by suitecrm core.
+            $this->logger->fatal(__FILE__ . ':' . __LINE__ . ' ' . __METHOD__ . ' - ' . 'More than one meeting matches Google Id: ' . $eventIdQuoted);
+            throw new GoogleSyncException('More than one meeting matches Google Id!', GoogleSyncException::AMBIGUOUS_MEETING_ID);
+        } elseif ($result->num_rows == 0) {
+            // TODO: Confusing return value: Do not use similar return value by different reason of failure. see belove the $meeting->retrive
+            return null; // No matches Found
+        }
+
+        $meeting = BeanFactory::getBean('Meetings');
+        if (!$meeting) {
+            throw new GoogleSyncException('Unable to get Meetings bean.', GoogleSyncException::UNABLE_TO_RETRIEVE_MEETING);
+        }
+        $row = $this->db->fetchByAssoc($result);
+        $meeting->retrieve($row['id'], true, false); // retrieve() method could failing, but it's not handled.
+
+        return $meeting;
+    }
+
+    /**
+     * Creates and Sets $this->gService to a valid Google Calendar Service
+     *
+     * @return bool Success/Failure
+     */
+    protected function setGService()
+    {
+        // make sure we have a client set
+        if (!isset($this->gClient)) {
+            // TODO: Confusing return value: It should be an exception, do not use similar return value (false) by different reaon os failure (see belove)
```

@gymad 
I think you're right. It should throw an exception. Will fix.

---

Fixed